.github/workflows/ci.yaml

```diff
@@ -23,80 +25,57 @@ jobs:
         with:
           test-results: test.json
 
   test:
     runs-on: ubuntu-latest
-    # Define any services needed for the test suite (or delete this section)
-    # services:
-    #   postgres:
-    #     image: postgres:16
-    #     ports:
-    #       - "5432:5432"
-    #     env:
-    #       POSTGRES_PASSWORD: secretpassword
     env:
       BATON_LOG_LEVEL: debug
-      # Add any environment variables needed to run baton-paylocity
-      # BATON_BASE_URL: 'http://localhost:8080'
-      # BATON_ACCESS_TOKEN: 'secret_token'
-      # The following parameters are passed to grant/revoke commands
-      # Change these to the correct IDs for your test data
-      CONNECTOR_GRANT: 'grant:entitlement:group:1234:member:user:9876'
-      CONNECTOR_ENTITLEMENT: 'entitlement:group:1234:member'
-      CONNECTOR_PRINCIPAL: 'user:9876'
-      CONNECTOR_PRINCIPAL_TYPE: 'user'
+      BATON_PAYLOCITY_CLIENT_ID: test-client-id
+      BATON_PAYLOCITY_CLIENT_SECRET: test-client-secret
+      BATON_PAYLOCITY_COMPANY_ID: test-company
+      BATON_PAYLOCITY_BASE_URL: http://localhost:8080
+
     steps:
       - name: Install Go
         uses: actions/setup-go@v5
         with:
           go-version: 1.23.x
       - name: Checkout code
         uses: actions/checkout@v4
-      # Install any dependencies here (or delete this)
-      # - name: Install postgres client
-      #   run: sudo apt install postgresql-client
-      # Run any fixture setup here (or delete this)
-      # - name: Import sql into postgres
-      #   run: psql -h localhost --user postgres -f environment.sql
-      #   env:
-      #     PGPASSWORD: secretpassword
-      - name: Build baton-paylocity
-        run: go build ./cmd/baton-paylocity
-      - name: Run baton-paylocity
-        run: ./baton-paylocity
-
-      - name: Install baton
-        run: ./scripts/get-baton.sh && mv baton /usr/local/bin
 
-      - name: Check for grant before revoking
 
-        run:
-          baton grants --entitlement="${{ env.CONNECTOR_ENTITLEMENT }}" --output-format=json | jq --exit-status ".grants[].principal.id.resource == \"${{ env.CONNECTOR_PRINCIPAL }}\""
+      - name: Build test server
+        run: go build -o test-server ./cmd/test_server
+      - name: Start test server
+        run: ./test-server &
+      - name: Wait for test server to start
+        run: sleep 5
 
-      - name: Revoke grants
-        run: ./baton-paylocity --revoke-grant="${{ env.CONNECTOR_GRANT }}"
+      - name: Get baton
+        uses: ConductorOne/github-workflows/actions/get-baton@v2
 
-      - name: Check grant was revoked
-        run: ./baton-paylocity && baton grants --entitlement="${{ env.CONNECTOR_ENTITLEMENT }}" --output-format=json | jq --exit-status "if .grants then .grants[]?.principal.id.resource != \"${{ env.CONNECTOR_PRINCIPAL }}\" else . end"
+      - name: Build baton-paylocity
+        run: go build ./cmd/baton-paylocity
+      - name: Run baton-paylocity (sync)
```

Comment:
There's also a github action to run basic sync tests: https://github.com/ConductorOne/github-workflows/tree/main/actions/sync-test

You can find some example usages here: https://github.com/ConductorOne/baton-http/blob/main/.github/workflows/ci.yaml#L100

It looks for a specific grant, not just a count. Also if the connector supports provisioning, it will test granting/revoking.